pkg/analysis/nomaps/doc.go

```diff
@@ -1,11 +1,24 @@
 /*
-nophase provides a linter to ensure that structs do not contain a Phase field.
+nomaps provides a linter to ensure that structs do not contain a map field.
 
-Phase fields are deprecated and conditions should be preferred. Avoid phase like enum
-fields.
+Map fields are discouraged in Kubernetes API types. Use a list of key-value pairs.
 
-The linter will flag any struct field containing the substring 'phase'. This means both
-Phase and FooPhase will be flagged.
+The following struct isn't recommended
+
+type Spec struct {
+	Pairs map[string]string
+}
+
+Instead, use a list of key-value pairs
+
+type Spec struct {
+	Pairs []Pair
+}
+
+type Pair struct {
+	Key   string
+	Value string
+}
```

Comment:
This example isn't correct, since we allow `map[string]string`.

We need to focus the documentation more on avoiding maps of subobjects.

```
nomaps provides a linter to ensure that fields do not use map types.

Maps are discouraged in Kubernetes APIs. It is hard to distinguish between structs and maps in JSON/YAML and as such, lists of named subobjects are preferred over plain map types.

Instead of

  ports:
    www:
      containerPort: 80

use

  ports:
    - name: www
      containerPort: 80

Lists should use the `+listType=map` and `+listMapKey=name` markers, or equivalent.
```

Reply:
Fixed.